app/src/main/java/com/android/sample/model/camera/CameraViewModel.kt

```diff
@@ -100,12 +116,49 @@ open class CameraViewModel(private val context: Context) : ViewModel() {
                   if (event.hasError()) {
                     _recording.value?.close()
                     _recording.value = null
+                    resetCameraController()
                     onFailure()
                   } else {
+                    resetCameraController()
                     onSuccess()
                   }
                 }
               }
             }
   }
+
+  /** Switches the camera controller to the video capture use case. */
+  private fun switchVideoCaptureUseCase() {
+    _cameraController.value.setEnabledUseCases(CameraController.VIDEO_CAPTURE)
+  }
+
+  /** Resets the camera controller to its initial state with the front camera selected. */
+  private fun resetCameraController() {
+    _cameraController.value =
+        LifecycleCameraController(context).apply {
+          cameraSelector = CameraSelector.DEFAULT_FRONT_CAMERA
+        }
+  }
+
+  /** Enables pose recognition by setting up the image analysis analyzer. */
+  fun enablePoseRecognition() {
+    _cameraController.value.imageAnalysisTargetSize =
+        CameraController.OutputSize(AspectRatio.RATIO_16_9)
+    _cameraController.value.setImageAnalysisAnalyzer(
+        ContextCompat.getMainExecutor(context),
+        PoseDetectionAnalyser(
+            onDetectedPoseUpdated = {
+              Log.d("MLDEB", "enablePoseRecognition: ${poseLandmarks.value.size} ")
+              _poseLandMarks.value += it
+            }))
+  }
```

Comment:
Not sure if this is feasible but I've noticed that this method creates a new instance of PoseDetectionAnalyser() each time it's called, would there be a way to make sure that there's not already an instance of the analyser that maybe has not been cleared using the finishPoseRecognition() method ? Otherwise it might lead to duplication errors. However i think making sure to have at least one call to the finish method every time would be enough so my suggestion might be useless.

Reply:
Indeed, fiexed now